gpii/node_modules/testing/src/Integration.js

```diff
@@ -369,3 +372,57 @@ fluid.defaults("gpii.test.integration.mockSettingsHandlerRegistry.android", {
         }
     }
 });
+
+fluid.defaults("gpii.test.integration.mockDeviceReportersRegistry", {
+    gradeNames: ["fluid.eventedComponent", "autoInit"],
+    members: {
+        deviceReporters: {}
+    },
+    listeners: {
+        onCreate: "gpii.test.integration.mockDeviceReportersRegistry.populate"
+    }
+});
+
+gpii.test.integration.mockDeviceReporters = {};
+
+gpii.test.integration.mockDeviceReporters.findNameInList = function (expectInstalled, name) {
+    return expectInstalled.some(function (el) { return el === name; });
+};
+
+gpii.test.integration.mockDeviceReportersRegistry.populate = function (that) {
+    fluid.each(that.options.deviceReporters, function (options, key) {
+        var expectInstalled = [];
+        fluid.each(gpii.tests, function (tests) {
```

Comment:
@amb26,

in response to this comment, I've made a few changes in https://github.com/javihernandez/universal/commit/b9b8ee99edaf8a30078e5928a980167c1e02e5b9

Not sure if I've fixed the race condition you mention before, but at least, it works as expected, including the tests.